docs/tutorials/clojure-guide.md

```diff
@@ -0,0 +1,174 @@
+
+## Installing the language server
+
+`lsp-mode` is a client-server application with `Emacs` acting as the client. In
+order for it to work, you have to install a separate _language server_ that
+understands the specifics of your target language. 
+
+For Clojure, we will use [clojure-lsp](https://github.com/clojure-lsp/clojure-lsp) which as of 2021/01, is the only [LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/) 
+LSP server for `clojure`, it has a lot of useful features that I will try to show in this guide.
+
+### Via lsp-mode
+
+`lsp-mode` ships with automatic installation scripts for some of the language servers and clojure-lsp 
+is one of the supported servers, you can install it using <kbd>M-x</kbd> `lsp-install-server` <kbd>RET</kbd> `clojure-lsp`. 
+After installed, `lsp-mode` will automatically initialize it when opening clojure files.
+
+### Manually
+
+Check [here](https://github.com/clojure-lsp/clojure-lsp#installation) for more information on how to build it manually.
+
+For a custom path, you can set the path of the server location with:
+
+```elisp
+(setq lsp-clojure-custom-server-command '("bash" "-c" "/path/to/clojure-lsp"))
+```
+
+## LSP features
+
+`lsp-mode` has a lot of features and integrations, this guide will show most important ones 
+that I think it's important on clojure development. You can check all the features on the 
+official [lsp-mode documentation](https://emacs-lsp.github.io/lsp-mode/)
+
+For more info on `lsp-mode` features, check [here](https://emacs-lsp.github.io/lsp-mode/page/main-features/).
+
+### Syntax check
+
+It checks for errors and warnings on your code while you write it, it uses [clj-kondo](https://github.com/clj-kondo/clj-kondo), 
+another amazing library that lints clojure code, in the back-end so it's not required to install `clj-kondo` since `clojure-lsp` will
+use it automatically.
+
+It will also check for public unused functions/variables which is something as far as I known that does not exists in 
+any other clojure library/package:
+
+![syntax-checker](images/clojure-syntax-checker.png "syntax-checker")
+
+### Find definition/references
+
+You can find a function definition with `lsp-find-definition` or find all the references to that 
+symbol, variable or function with `lsp-find-references` or `lsp-ui-peek-find-references` if 
+[lsp-ui](https://emacs-lsp.github.io/lsp-ui/) is available. 
+
+![find-references](images/clojure-find-references.gif "find-references")
+
+<details>
+<summary>Doom emacs users</summary>
+
+You may need to remove the lookup handlers conflict from `cider` and/or `clj-refactor` if you want to use this LSP feature.
+
+```elisp
+(use-package! cider
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'cider-mode nil))
+  
+(use-package! clj-refactor
+  :after clojure-mode
+  :config
+  (set-lookup-handlers! 'clj-refactor-mode nil))
+```
+
+</details>
```

Comment:
It'd be nice to have a section like that for Spacemacs users

Reply:
`use-package` works on Spacemacs, so if `use-package!` -> `use-package`, everything should work (untested though).